arm-compute/2016-09-30/containerService.json

```diff
@@ -0,0 +1,561 @@
+          "202": {
+            "description": ""
+          },
+          "204": {
+            "description": ""
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/containerServices": {
+      "get": {
+        "tags": [
+          "ContainerService"
+        ],
+        "operationId": "ContainerService_List",
+        "description": "The operation to list container services.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/ContainerServiceListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    }
+    },
+    "definitions": {
+        "Resource": {
+          "properties": {
+            "id": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource Id"
+            },
+            "name": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource name"
+            },
+            "type": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource type"
+            },
+            "location": {
+              "type": "string",
+              "description": "Resource location"
+            },
+            "tags": {
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              },
+              "description": "Resource tags"
+            }
+          },
+          "required": [
+            "location"
+          ],
+          "x-ms-azure-resource": true
+        },
+        "ContainerServiceCustomProfile": {
+          "properties": {
+            "orchestrator": {
+              "type": "string",
+              "description": "The name of the custom orchestrator to use"
+            }
+          },
+          "required": [
+            "orchestrator"
+          ],
+          "description": "Properties for customizing this cluster"
+        },
+        "ContainerServiceServicePrincipalProfile": {
+          "properties": {
+            "clientId": {
+              "type": "string",
+              "description": "The id for this service principal"
+            },
+            "clientSecret": {
+              "type": "string",
+              "description": "The secret associated with this service principal"

```

Comment:
Currently service principals support  a secret/password and a certificate. So does the service expect the user to provide anything over here? If yes or no, then can you please document the correct behavior over here